src/Cache/InMemoryCache.ts

```diff
@@ -3,18 +3,21 @@ import { ICache } from './ICache';
 export class InMemoryCache implements ICache {
 	private cache = {};
 
-	public set(key: string, value: any): void {
+	public set(key: string, value: any): any {
 		this.cache[key] = value;
+		return value !== null && typeof value === 'object' ? Object.assign({}, value) : value;
```

Comment:
What about approach that se does the cloning and then stores the cloned value and return type will be still void.
I think that this would be save becouse with current aproach user of cache has to remember that he must use returned value...